src/SignalR/server/Core/src/DefaultHubLifetimeManager.cs

```diff
@@ -238,7 +238,26 @@ public override Task SendGroupsAsync(IReadOnlyList<string> groupNames, string me
             var group = _groups[groupName];
             if (group != null)
             {
-                DefaultHubLifetimeManager<THub>.SendToGroupConnections(methodName, args, group, null, null, ref tasks, ref message, cancellationToken);
+                foreach (var connection in group)
+                {
+                    if (connections == null)
+                    {
+                        connections = new HashSet<string>();
+                    }
+                    connections.Add(connection.Key);
+                }
+            }
+        }
+
+        if (connections != null)
+        {
+            foreach (var connectionId in connections)
+            {
+                if (tasks == null)
+                {
+                    tasks = new List<Task>();
+                }
```

Comment:
HashSet has a `Count` property, so initialize the `tasks` list to proper size before the loop, thus avoiding this `if`.